server/migrations/helpers.js

```diff
@@ -1,50 +1,62 @@
 function calculatePreferences(user) {
-  const updatedPreferences = {
-    config: '',
-    sort: 0,
-    star: user?.preferences?.star || [],
-    complete: user?.preferences?.complete || [],
-  }
-  // Update complete
-  if (
-    !!user.preferences &&
-    user.preferences.complete &&
-    Object.keys(user.preferences.complete).length &&
-    !Array.isArray(user.preferences.complete)
-  ) {
-    updatedPreferences.complete = Object.keys(user.preferences.complete).map(
-      (siteName) => {
-        return {
-          site: siteName,
-          completedSubjects: user.preferences.complete[siteName],
-        }
+  const siteSubjects = []
+
+  if (!user.siteSubjects?.length) {
+    user.access.forEach((site, i) => {
+      const { preferences } = user
+      const siteSubject = {
+        site,
+        starredSubjects: preferences.star[site]?.length
+          ? preferences.star[site]
+          : [],
+        completedSubjects: preferences.complete[site]?.length
+          ? preferences.complete[site]
+          : [],
       }
-    )
+      siteSubjects.push(siteSubject)
+    })
   }
+  if (user.siteSubjects && user.siteSubjects.length) {
+    user.access.forEach((study, i) => {
+      const siteSubject = {
+        site: study,
+        starredSubjects: [],
+        completedSubjects: [],
+      }
+
+      if (user.siteSubjects.findIndex(({ site }) => study === site) > -1) {
+        return siteSubjects.push(
+          user.siteSubjects[
+            user.siteSubjects.findIndex(({ site }) => study === site)
+          ]
+        )
```

Comment:
I'm confused about what we're doing here. In the condition above we determine that the user already has site subjects. That suggests to me that this user has already been migrated because users didn't have site subjects before this change, is that correct? If they've already been migrated, do we need to do anything to the user?

Reply:
This was a just in case we missed something during the user's migration it would just redo it, but yes it looks redundant.

Reply:
https://github.com/AMP-SCZ/dpdash/pull/241/commits/bf0423c6c7f4d2b24d8c720cb39b4accdaccbbc7